src/plots/cartesian/graph_interact.js

```diff
@@ -129,6 +144,11 @@ fx.init = function(gd) {
             maindrag.onmouseout = function(evt) {
                 if(gd._dragging) return;
 
+                // When the mouse leaves this maindrag, unset the hovered subplot.
+                // This may cause problems if it leaves the subplot directly *onto*
+                // another subplot, but that's a tiny corner case at the moment.
+                gd._fullLayout._hoversubplot = null;
```

Comment:
Yes, which happens first was the question. If mouseover and then mouseout, it would end up unset, but the *only* time that could happen is if you move *exactly* one pixel outside of the inset region, which is where it started to seem like a small corner case. The second pixel moved will immediately fix things.